### PR TITLE
INT - B-21481 -  DB -  Add HI/AK duty locations

### PR DESCRIPTION
**Pull request by MInthavongsay**
## Summary

This PR introduces a new database migration file to insert 533 Hawaii and Alaska duty locations. A total of 533 duty location records were created out of possible 639 from the provided TRDM file. 106 duty locations were ignored/excluded per TRDM's instructions(flag).   

Feature flag -- enable_alaska -  has been been incorporated to include/exclude both HI and AK duty locations from duty location search results. This feature serves as a transitional feature to bridge the work between data migration and the issue related to empty counseling office dropdown. Another backlog ticket will be created to address the counseling office issue. This FF enables this ticket to be fulfilled per AC.

Note: This PR contains [DB - INT PR 21442](https://github.com/transcom/mymove/pull/13872)

**New Migration File: migrations/app/schema/20241017183144_add_AK_HI_duty_locations.up.sql**

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Edit .envrc and set FEATURE_FLAG_ENABLE_ALASKA to TRUE. Run direnv allow. Login as customer/office. Goto view containing Duty Location widget and search for Alaska and Hawaii locations....ie Honolulu, Anchorage ...etc... See attached file below for 599 expected locations. Confirm AK and HI locations are present in the results.
2. Edit .envrc and set FEATURE_FLAG_ENABLE_ALASKA to FALSE. Run direnv allow. Login as customer/office. Goto view containing Duty Location widget and search for Alaska and Hawaii locations....ie Honolulu, Anchorage ...etc... See attached file below for 599 expected locations. Confirm AK and HI locations are not present in the results.

**599 Duty Locations:**
[HI-AK-data-1729264016473.csv](https://github.com/user-attachments/files/17462373/HI-AK-data-1729264016473.csv)

Note: 
106 Excluded Duty Locations:
[null-us_post_region_cities_id.csv](https://github.com/user-attachments/files/17462406/null-us_post_region_cities_id.csv)

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: migrations/app/schema/20241017183144_add_AK_HI_duty_locations.up.sql [hunk 1]
```diff
@@ -0,0 +1,1069 @@
+-- Insert associated records for HI and AK duty location. There are no specific address but will use placeholder address with 'n/a' for street address 1 using known city, state, zip and county.
+INSERT INTO public.addresses VALUES ('4d1f305a-fb7d-4191-ab3c-349e749d340a','n/a',null,'KAILUA KONA','HI','96745','2024-10-17 18:12:00.374399','2024-10-17 18:12:00.374399',null,'HAWAII',true,'791899e6-cd77-46f2-981b-176ecb8d7098','345360d0-c3e9-4d87-b60b-59ab4b50c9ff');
```
Comment (r-mettler) [inline migrations/app/schema/20241017183144_add_AK_HI_duty_locations.up.sql:2]: We should change these to use the COPY command, its best for large data inserts. I can help you bulk edit this in VSCode if you need to.
See the `20241004151509_insert_re_cities.up.sql` file for an example on the format. Note those are tabs in between the data.

Comment (MInthavongsay) [reply migrations/app/schema/20241017183144_add_AK_HI_duty_locations.up.sql]: Done.

File: .envrc [hunk 2]
```diff
@@ -159,6 +159,9 @@ export FEATURE_FLAG_UNACCOMPANIED_BAGGAGE=false
 # Feature flag to allow AK to be entered as a state
 export FEATURE_FLAG_ENABLE_ALASKA=false
 
+# Feature flag to allow AK to be entered as a state
```
[the remark this answers is not in the document]
Comment (MInthavongsay) [reply .envrc:162]: This is probably coming from the merge of 21442.

[the remark this answers is not in the document]
Comment (MInthavongsay) [reply .envrc:162]: It looks like it came from original merge of B-21442-INT into my INT branch. Merging latest of B-21442-INT corrected the issue.

**Review by ajlusk — APPROVED** [review]
No AK & HI locations with the feature flag off:
Screenshot 2024-10-29 at 11 12 27 AM

AK & HI flags with the feature flag on:
Screenshot 2024-10-29 at 11 07 00 AM
Screenshot 2024-10-29 at 11 06 25 AM

No Counseling Offices available when an AK or HI duty location is selected as expected: 
Screenshot 2024-10-29 at 11 27 41 AM

"City Names To Avoid" are not included
Screenshot 2024-10-29 at 11 07 52 AM

**Comment by ajlusk** [comment]
Unrelated to these changes but I did notice even with enable_alaska FF set to true I am not able to use an AK zip code in my customer's current address in the initial account set up:
Screenshot 2024-10-29 at 11 05 23 AM

However, I am able to edit my existing current address to an AK zip code:
Screenshot 2024-10-29 at 11 32 24 AM

First part may be something that the alaska FF needs to include. Second part seems like an oversight from at least 16 months ago.

**Review by begrohmann — APPROVED** [review]

**Review by traskowskycaci — CHANGES_REQUESTED** [review]
Test coverage for pkg/handlers/internalapi/duty_locations.go is showing lots of red, can you add a test in? Here's an example of how to mock the feature flag fetcher: https://github.com/transcom/mymove/pull/13976/files#diff-eed993d434b4dccb23f242b5fe4737faceabef22a440321423563a2ff454e548

image

**Comment by traskowskycaci** [comment]
> Test coverage for pkg/handlers/internalapi/duty_locations.go is showing lots of red, can you add a test in? Here's an example of how to mock the feature flag fetcher: https://github.com/transcom/mymove/pull/13976/files#diff-eed993d434b4dccb23f242b5fe4737faceabef22a440321423563a2ff454e548
> 


You're using GetBooleanFlagForUser, so here is a file with a more closely related example on how to mock that - https://github.com/transcom/mymove/blob/integrationTesting/pkg/handlers/internalapi/mto_shipment_test.go

**Review by traskowskycaci — APPROVED** [review]
Thanks for adding those tests in!

**Comment by traskowskycaci** [comment]
> This is unrelated to these changes but I did notice even with enable_alaska FF set to true I am not able to use an AK zip code in my customer's current address in the initial account set up: Screenshot 2024-10-29 at 11 05 23 AM
> 
> However, I am able to edit my existing current address to an AK zip code: Screenshot 2024-10-29 at 11 32 24 AM
> 
> First part may be something that the alaska FF needs to include. Second part seems like an oversight from at least 16 months ago.

That error message and AK zip codes will be handled with @r-mettler's [B-21168](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1011663) work

**Pull request merged**